rust/processor/src/gap_detectors/gap_detector.rs

```diff
@@ -24,7 +20,10 @@ pub struct DefaultGapDetectorResult {
 }
 
 impl GapDetectorTrait for DefaultGapDetector {
-    fn process_versions(&mut self, result: ProcessingResult) -> anyhow::Result<GapDetectorResult> {
+    fn process_versions(
+        &mut self,
+        result: ProcessingResult,
+    ) -> anyhow::Result<GapDetectorResult, Box<dyn std::error::Error + Send + Sync>> {
```

Comment:
How come this error type is required? 

If it is required, might as well just use `Result`, `anyhow::Result` is just an alias for a result where the error type is `anyhow::Error`.